tests/test_result_manager.py

```diff
@@ -132,45 +129,44 @@ def test_get_single_model_sorted_results(self):
         with only results for that model 
         """
         result_manager, _ = load_single_model_result_manager()
-        result_manager.add_run_config_measurement(MagicMock(), MagicMock())
-        result_manager.compile_and_sort_results()
+        result_manager._add_rcm_to_results(MagicMock(), MagicMock())
         sorted_results = result_manager.get_model_sorted_results("add_sub")
         self.assertTrue(isinstance(sorted_results, ResultHeap))
         self.assertEqual(5, len(sorted_results.results()))
 
     def test_get_multi_model_sorted_results(self):
         """ Test get_model_sorted_results returns a valid ResultHeap """
         result_manager, _ = load_multi_model_result_manager()
-        result_manager.compile_and_sort_results()
         sorted_results = result_manager.get_model_sorted_results(
             "resnet50_libtorch,vgg19_libtorch")
         self.assertTrue(isinstance(sorted_results, ResultHeap))
         self.assertEqual(17, len(sorted_results.results()))
 
-    def test_get_across_model_sorted_results(self):
-        """ 
-        Test get_across_model_sorted_results returns a valid ResultHeap
-        with all results across all models 
-        """
-        result_manager, _ = load_single_model_result_manager()
-        self._add_a_fake_result(result_manager)
-        result_manager.compile_and_sort_results()
-
-        sorted_results = result_manager.get_across_model_sorted_results()
-        self.assertTrue(isinstance(sorted_results, ResultHeap))
-        self.assertEqual(6, len(sorted_results.results()))
-
-    def _add_a_fake_result(self, result_manager):
-        fake_model = MagicMock()
-        fake_model.model_name.return_value = "FakeModel"
-        old_analysis_models = result_manager._config.analysis_models
-        old_analysis_models.append(fake_model)
-        result_manager._config.analysis_models = old_analysis_models
-
-        fake_run_config = MagicMock()
-        fake_run_config.models_name.return_value = "FakeModel"
-
-        result_manager.add_run_config_measurement(fake_run_config, MagicMock())
+    # FIXME: This doesn't work anymore since model names are loaded at __init__
```

Comment:
@tgerdesnv - This unit test is problematic and I can't think of a good way of easily fixing it. The issue is that all models must now be known when the result manager is created. Thoughts?

Reply:
We need to be able to say "get top 3 configs across all models". Is this protected anywhere else? Does that functionality still work?

Reply:
I've fixed the unit test